api/src/main/java/org/openmrs/module/registrationcore/api/mpi/openempi/PatientBuilder.java

```diff
@@ -88,24 +89,24 @@ private void setMpiPatientIdentifiers(OpenEmpiPatientResult patientQuery, Patien
             String mpiIdentifierTypeName = identifier.getIdentifierDomain().getIdentifierDomainName();
             Integer mpiIdentifierTypeId = identifier.getIdentifierDomain().getIdentifierDomainId();
 
-            Integer localIdentifierTypeId = identifierMapper.getMappedLocalIdentifierTypeId(mpiIdentifierTypeId);
+            String localIdentifierTypeUuid = identifierMapper.getMappedLocalIdentifierTypeUuid(mpiIdentifierTypeId.toString());
             String identifierValue = identifier.getIdentifier();
-            PatientIdentifier patientIdentifier = createIdentifier(mpiIdentifierTypeName, localIdentifierTypeId, identifierValue);
+            PatientIdentifier patientIdentifier = createIdentifier(mpiIdentifierTypeName, localIdentifierTypeUuid, identifierValue);
 
             patient.addIdentifier(patientIdentifier);
         }
     }
 
     private void setMpiPersonIdentifier(OpenEmpiPatientResult patientQuery, Patient patient) {
-        Integer personIdentifierId = mpiProperties.getMpiPersonIdentifierTypeId();
-        PatientIdentifier identifier = createIdentifier("Person identifier", personIdentifierId, String.valueOf(patientQuery.getPersonId()));
+        String personIdentifierTypeUuid = mpiProperties.getMpiPersonIdentifierTypeUuid();
+        PatientIdentifier identifier = createIdentifier("Person identifier", personIdentifierTypeUuid, String.valueOf(patientQuery.getPersonId()));
         patient.getIdentifiers().add(identifier);
     }
 
-    private PatientIdentifier createIdentifier(String identifierName, Integer identifierId, String identifierValue) {
-        log.info("Create identifier for imported Mpi patient. Identifier name: " + identifierName + ". Identifier Id: "
-                + identifierId + ". Identifier value: " + identifierValue);
-        PatientIdentifier identifier = identifierBuilder.createIdentifier(identifierId, identifierValue, null);
+    private PatientIdentifier createIdentifier(String identifierName, String identifierTypeUuid, String identifierValue) {
+        log.info("Create identifier for imported Mpi patient. Identifier name: " + identifierName + ". Identifier UUID: "
+                + identifierTypeUuid + ". Identifier value: " + identifierValue);
+        PatientIdentifier identifier = identifierBuilder.createIdentifier(identifierTypeUuid, identifierValue, null);
         if (OPENMRS_IDENTIFIER_NAME.equals(identifierName)) {
```

Comment:
This should be UUID i think and not name

Reply:
Changed